packages/react-native-payments/lib/js/helpers/index.js

```diff
@@ -118,7 +118,7 @@ export function validateTotal(total, errorType = Error): void {
     throw new errorType(`required member total is undefined.`);
   }
 
-  const hasTotal = total && total.amount && total.amount.value;
+  const hasTotal = total && total.amount && (total.amount.value || total.amount.value === 0)
```

Comment:
I would add `@flow` comment to top and proper typing for `const hasTotal`

 ```
const hasTotal: boolean = total && total.amount && total.amount.value >= 0;
```

Reply:
by checking `total.amount.value >= 0` you can get rid of `isNegative` check later in this function